src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -873,10 +874,32 @@ protected virtual void Dispose(bool disposing)
 
         public void Dispose()
         {
+            StopWorkerProcessAsync();
```

Comment:
can you move this under setting `_disposing = true;`

Reply:
Little unsure of that as `StopWorkerProcessAsync( )` is trying to inform the worker about termination while actual `dispose()` is starting after that. I think we can follow same pattern as `StartWorkerProcessAsync()` where it tells the worker first and then set the state.

https://github.com/Azure/azure-functions-host/blob/9a81e05c957e492c8887a8c2cd180d927f594b35/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L155-L156